ethcore/src/state/account.rs

```diff
@@ -145,8 +145,14 @@ impl Account {
 
 	/// Create a new account from RLP.
 	pub fn from_rlp(rlp: &[u8]) -> Account {
-		let basic: BasicAccount = ::rlp::decode(rlp);
-		basic.into()
+		match ::rlp::decode::<BasicAccount>(rlp) {
+			Ok(basic_account) => basic_account.into(),
+			Err(e) => {
+				// REVIEW: Is panicking ok here? I'd like to change the return value here to `Result<Account, SomeError>` but not sure if that's desirable or the right time.
```

Comment:
I believe it's not ok, cause we don't know where decoded slice of bytes comes from. It's better to change it to `Result<Account, SomeError>`